Group member list screen UI

divy-arpit opened the pull request:
### Fixes #44 

### Description

This PR adds to the app a screen which would show the members of a group

**Type of Change:** (Delete irrelevant options)
- Code
- User Interface


**Code/Quality Assurance Only** (Delete irrelevant options)
- New feature (non-breaking change which adds functionality pre-approved by mentors)


### How Has This Been Tested?

Tested on local device.


### Checklist:

- [X] My PR follows the style guidelines of this project
- [X] I have performed a self-review of my own code or materials
- [X] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged


**Code/Quality Assurance**
- [ ] My code follows TDD clean architecture
- [X] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

![Screenshot_1591164310](https://user-images.githubusercontent.com/61823356/83602047-6a447880-a58f-11ea-91e2-5c41753ab73d.png)

divy-arpit committed:
Add a screen to display the members of a group

This commit adds the UI which would show a list of members of the group.

divy-arpit committed:
Indent

divy-arpit committed:
Add a screen to display the members of a group

This commit adds the UI which would show a list of members of the group.

divy-arpit committed:
Add a floatingActionButton

naveennvrgup commented on lib/screens/group_member_list/widgets/create_list_tile.dart:
```diff
@@ -0,0 +1,27 @@
+///This class returns a list tile which would hold the name and details of member
+///It receives a string as parameter which is the name of the person
+
+class ListTileCreate extends StatelessWidget {
```
```suggestion
class GroupMemberTile extends StatelessWidget {
```

naveennvrgup replied:
also please rename the file accordingly

naveennvrgup commented on lib/screens/group_member_list/widgets/create_list_tile.dart:
```diff
@@ -0,0 +1,27 @@
+class ListTileCreate extends StatelessWidget {
+  final String title;
+
+  ListTileCreate(this.title);
```
1. Please take in profilePicUrl(optional) and memberId(compulsory) also. 
2. If the profilePicUrl is null then show account_circle else pic from that URL
3. Use named arguments :innocent:

naveennvrgup reviewed (changes requested):
1. Great job with PR @divy-arpit  :fire:
1. nice to see PR from a separate branch

divy-arpit committed:
Add profilePictureUrl and improve file naming

This commit adds a profilePictureUrl which would be used to display the profile picture and improves the naming of classes and files

naveennvrgup commented on pubspec.yaml:
```diff
@@ -28,6 +28,7 @@ dependencies:
   google_fonts: ^0.5.0+1
   # Picking images from the image library
   image_picker: ^0.6.6+5
+  transparent_image: ^1.0.0
```
I think it provides redudant functionality. I suggest you try the flutter's built in fadeinimg widget.  
https://api.flutter.dev/flutter/widgets/FadeInImage-class.html
```
      return FadeInImage(
        placeholder: AssetImage("assets/images/icon_ecell.png"),
        image: NetworkImage(this.profilePicUrl),
      );
```

naveennvrgup commented on lib/screens/group_member_list/widgets/group_member_tile.dart:
```diff
@@ -0,0 +1,39 @@
+class GroupMemberTile extends StatelessWidget {
+  final String title, profilePicUrl;
+
+  GroupMemberTile({@required this.title, this.profilePicUrl});
```
sorry to mention it late you can take in member id also :sweat_smile:

divy-arpit replied:
I assume that member ID is going to be an integer and so I have added functionalities accordingly sir.

divy-arpit committed:
Add memberId and remove transparent_image for pubspec.yaml

This commit adds memberId and removes the transparent_image from pubspec.yaml. Also, an image for the placeholder has been added.
